src/main/java/io/vlingo/symbio/store/object/jdbc/JDBCObjectStoreActor.java

```diff
@@ -69,13 +70,13 @@ public JDBCObjectStoreActor(final JDBCObjectStoreDelegate delegate, final Dispat
     final long timeout = delegate.configuration.transactionTimeoutMillis;
     stage().scheduler().schedule(selfAs(Scheduled.class), null, timeout, timeout);
 
+    this.dispatchControlDelegate = delegate.copy();
```

Comment:
There should be no reason to retain a copy of the delegate used by the `DispatcherControl`. Actors should be designed to share nothing. (I do break this rule in the in-memory examples because the backing store is a single in-memory collection.)

Reply:
The `DispatcherControl` is created inside the `JDBCObjectStoreActor` and it's hidden from the client using the store.  The `JDBCObjectStoreDelegate`  delegate implements both the store and dispatcher control protocols, serving for both actors. That's why I have to create a copy of it and also keep the reference for registering the mapper.

Reply:
I understand, but if you see the code at this link you will see that the `JDBCObjectStoreActor` is modifying the `JDBCObjectStoreDelegate` state that is actually owned by the `DispatcherControlActor`. The state owned by one actor should never be shared with another actor, for one because it can cause race conditions on that state. I do see the dilemma now. The `DispatcherControlActor` is common and doesn't understand mappers, so registering the mapper with the `DispatcherControlActor` doesn't make sense, and thus you want to modify the `JDBCObjectStoreDelegate`. There should be another way.

Reply:
What do you think about removing the `registerMapper()` from `ObjectStore` and leaving it solely for delegate. We could have a new contract for delegate, for example `ObjectStoreDelegate`. The actor has no reason to know about internals of the storage delegate.

Reply:
Seems like a good choice. The mapper would have to be set on the concrete delegate before creating/starting the `JDBCObjectStoreActor`.

Reply:
Because the ObjectStore will have to know how to persist the `Dispatchable` in the same transaction,  it makes sene to have the same delegate implementation for both `ObjectStore` and `DispatchControllActor` with mapping. When creating the `DispatchControllActor` one will have to copy the delegate to use other connection.

Reply:
Seems good. It doesn't seem like `copy()` must do a deep copy as in copying the mappings. The 2+ delegate copies can reference the same mappings. Agreed?

Reply:
Agree